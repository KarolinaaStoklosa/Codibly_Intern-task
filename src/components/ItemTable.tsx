import React from 'react';
import { ItemProps } from '../helpers/interfaces';
import {  TableRow, TableCell, Modal, Box, Typography } from '@mui/material';

const modalStyle = {
    position: 'absolute' as 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    width: 400,
    bgcolor: 'background.paper',
    border: '2px solid #000',
    boxShadow: 24,
    p: 4,
  };
    

const ItemTable: React.FC<ItemProps> = (item, key) => {

    const {id, name, year, color, pantone_value} = item.item
    modalStyle.bgcolor = color

    const [open, setOpen] = React.useState(false);
    const handleOpen = () => setOpen(true);
    const handleClose = () => setOpen(false);

    return (
        <>
            <TableRow data-testid="table-row"
                key={key} sx={{bgcolor: color}} onClick={handleOpen}
                >
                <TableCell component="th" scope="row" >{id}</TableCell>
                <TableCell>{name}</TableCell>
                <TableCell>{year}</TableCell>
            </TableRow>
            <Modal
                data-testid="modal"
                open={open}
                onClose={handleClose}
                aria-labelledby="modal-modal-title"
                aria-describedby="modal-modal-description"
                >
                <Box sx={modalStyle}>
                    <Typography id="modal-modal-title" variant="h6" component="h2">
                        id: {id}
                    </Typography>
                    <Typography id="modal-modal-description" sx={{ mt: 2 }}>
                        name: {name}
                    </Typography>
                    <Typography id="modal-modal-description" sx={{ mt: 2 }}>
                        year: {year}
                    </Typography>
                    <Typography id="modal-modal-description" sx={{ mt: 2 }}>
                        color: {color}
                    </Typography>
                    <Typography id="modal-modal-description" sx={{ mt: 2 }}>
                        pantone value: {pantone_value}
                    </Typography>
                </Box>
            </Modal>
        </>

    )
}

export default ItemTable;