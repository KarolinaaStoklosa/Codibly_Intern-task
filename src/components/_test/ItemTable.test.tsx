import { fireEvent, render, screen, getByTestId } from '@testing-library/react';
import ItemTable from '../ItemTable';

const itemValues = {
    id: 1, 
    name: 'red', 
    year: 2022, 
    color: '#FF0000', 
    pantone_value: '032 C'
}

describe("ItemTableTests", () => {
    it ("Table row shows correct values", () => {
        const { container } = render(
            <ItemTable 
                item={itemValues} key={1}
            />
        );
        expect(container).toMatchSnapshot();
    
    })

    it('clicking on row shows a modal', ()=> {
        const { getByTestId } = render(
            <ItemTable 
                item={itemValues} key={1}
            />
        );
        fireEvent.click(screen.getByTestId('table-row'))
        const elementAfterClick = screen.getByTestId('modal')
        expect(elementAfterClick).toBeInTheDocument();

    })
    it('modal contains pantone value', ()=> {
        const { getByTestId } = render(
            <ItemTable 
                item={itemValues} key={1}
            />
        );
        fireEvent.click(screen.getByTestId('table-row'))
        const elementAfterClick = screen.getByTestId('modal')
        expect(elementAfterClick).toHaveTextContent(/pantone value/i)
    })

})


