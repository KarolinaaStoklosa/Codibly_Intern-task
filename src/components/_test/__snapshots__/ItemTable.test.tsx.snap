// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemTableTests Table row shows correct values 1`] = `
<div>
  <tr
    class="MuiTableRow-root css-1i5bg5i-MuiTableRow-root"
    data-testid="table-row"
  >
    <th
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
      scope="row"
    >
      1
    </th>
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
    >
      red
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
    >
      2022
    </td>
  </tr>
</div>
`;
